webbpsf/optics.py

```diff
@@ -1184,12 +1191,18 @@ def __init__(self, instrument, include_oversize=False, **kwargs):
         self.instr_name = instrument.name
 
         # work out which name to index into the CV results with, if for NIRCam
+        is_nrc_coron = False
         if instrument.name == 'NIRCam':
             channel = instrument.channel[0].upper()
             lookup_name = "NIRCam{channel}W{module}".format(
                 channel=channel,
                 module=instrument.module
             )
+            # Check for coronagraphy
+            pupil_mask = self.instrument._pupil_mask
+            if (pupil_mask is not None) and ( ('LYOT' in pupil_mask.upper()) or ('MASK' in pupil_mask.upper()) ):
+                is_nrc_coron = True                
```

Comment:
[optional change] 
This can be simplified to: 
`is_nrc_coron = (pupil_mask is not None) and ( ('LYOT' in pupil_mask.upper()) or ('MASK' in pupil_mask.upper()) )`